pkg/commatrix-creator/commatrix.go

```diff
@@ -39,97 +40,125 @@ func New(exporter *endpointslices.EndpointSlicesExporter, customEntriesPath stri
 	if err != nil {
+		log.Errorf("Failed adding static entries: %s", err)
 		return nil, fmt.Errorf("failed adding static entries: %s", err)
 	}
 	epSliceComDetails = append(epSliceComDetails, staticEntries...)
 
 	if cm.customEntriesPath != "" {
+		log.Debug("Loading custom entries from file")
 		customComDetails, err := cm.GetComDetailsListFromFile()
 		if err != nil {
+			log.Errorf("Failed adding custom entries: %s", err)
 			return nil, fmt.Errorf("failed adding custom entries: %s", err)
 		}
 		epSliceComDetails = append(epSliceComDetails, customComDetails...)
 	}
 
 	commMatrix := &types.ComMatrix{Matrix: epSliceComDetails}
+	log.Debug("Cleaning ComDetails")
 	commMatrix.CleanComDetails()
 	return commMatrix, nil
 }
 
 func (cm *CommunicationMatrixCreator) GetComDetailsListFromFile() ([]types.ComDetails, error) {
+	log := utils.Logger() // Get the logger instance from utils
+
 	var res []types.ComDetails
+	log.Debugf("Opening file %s", cm.customEntriesPath)
 	f, err := os.Open(filepath.Clean(cm.customEntriesPath))
 	if err != nil {
+		log.Errorf("Failed to open file %s: %v", cm.customEntriesPath, err)
 		return nil, fmt.Errorf("failed to open file %s: %v", cm.customEntriesPath, err)
 	}
 	defer f.Close()
+
+	log.Debugf("Reading file %s", cm.customEntriesPath)
 	raw, err := io.ReadAll(f)
 	if err != nil {
+		log.Errorf("Failed to read file %s: %v", cm.customEntriesPath, err)
 		return nil, fmt.Errorf("failed to read file %s: %v", cm.customEntriesPath, err)
 	}
 
+	log.Debugf("Unmarshalling file content with format %s", cm.customEntriesFormat)
 	switch cm.customEntriesFormat {
 	case types.FormatJSON:
 		err = json.Unmarshal(raw, &res)
 		if err != nil {
+			log.Errorf("Failed to unmarshal JSON file: %v", err)
 			return nil, fmt.Errorf("failed to unmarshal custom entries file: %v", err)
 		}
 	case types.FormatYAML:
 		err = yaml.Unmarshal(raw, &res)
 		if err != nil {
+			log.Errorf("Failed to unmarshal YAML file: %v", err)
 			return nil, fmt.Errorf("failed to unmarshal custom entries file: %v", err)
 		}
 	case types.FormatCSV:
 		err = gocsv.UnmarshalBytes(raw, &res)
 		if err != nil {
+			log.Errorf("Failed to unmarshal CSV file: %v", err)
 			return nil, fmt.Errorf("failed to unmarshal custom entries file: %v", err)
 		}
 	default:
+		log.Errorf("Invalid format specified: %s", cm.customEntriesFormat)
 		return nil, fmt.Errorf("invalid value for format must be (json,yaml,csv)")
 	}
 
+	log.Debug("Successfully unmarshalled custom entries")
 	return res, nil
 }
 
 func (cm *CommunicationMatrixCreator) getStaticEntries() ([]types.ComDetails, error) {
+	log := utils.Logger() // Get the logger instance from utils
+
+	log.Debug("Determining static entries based on environment and deployment")
 	comDetails := []types.ComDetails{}
 
 	switch cm.e {
 	case types.Baremetal:
+		log.Debug("Adding Baremetal static entries")
 		comDetails = append(comDetails, types.BaremetalStaticEntriesMaster...)
-		if cm.d == types.SNO {
```

Comment:
Why is this change?

Reply:
mm because of the break i wrote on other way but its the same

Reply:
comDetails = append(comDetails, types.CloudStaticEntriesMaster...)
		if cm.d == types.SNO {
			break
			}
			comDetails = append(comDetails, types.CloudStaticEntriesWorker...)
	before we had that so i put itnsideon the if

Reply:
The idea of checking with equals is to make it clear in the code that this is what has to be done for a specific case. If we add more cases in the future we will have to rollback to equals.